### PR TITLE
adds access to response headers

### LINKED ISSUE #137: Having issue with reading HTTP headers for rate-limit
Using the API call `@lob.letters.create`.  Letter gets successfully sent out but not getting HTTP headers for rate-limit

```
@lob.letters.create(
  description: "Demo Letter",
  to: {
    name: "Harry Zhang",
    address_line1: "123 Test Street",
    address_city: "Mountain View",
    address_state: "CA",
    address_country: "US",
    address_zip: "94041"
  },
  from: {
    name: "Ami Wang",
    address_line1: "123 Test Street",
    address_city: "Mountain View",
    address_state: "CA",
    address_country: "US",
    address_zip: "94041"
  },
  file: "<html style='padding-top: 3in; margin: .5in;'>HTML Letter for {{name}}</html>",
  data: {
    name: "Harry"
  },
  color: true
)
```

### PR DESCRIPTION
**Pull request by mgartner**
Fixes #137 

Adds access to response headers by defining a singleton instance method on the response hash.

### CODE REVIEW

**Comment by charlesbjohnson** [comment]
LGTM

**Comment by charlesbjohnson** [comment]
Similar comment as https://github.com/lob/lob-node/pull/167

File: spec/lob_spec.rb [hunk 1]
```diff
@@ -62,4 +62,20 @@
...
     Lob.api_version = "2014-11-24"
   end
+
+  it "should include response headers" do
+    lob = Lob.load(api_key: ENV["LOB_API_KEY"])
+    result = lob.addresses.list
+    assert result.headers.include?(:content_type)
+  end
+
+  it "should handle API errors gracefully" do
```
Comment (robinjoseph08) [inline spec/lob_spec.rb]: `should include response headers for API errors`?

**Comment by robinjoseph08** [comment]
one thing but LGTM

**Comment by mgartner** [comment]
@charlesbjohnson updated!

**Comment by charlesbjohnson** [comment]
LGTM!

**Pull request merged**